File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-3.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String c : tokens) {
            switch(c){
                case "+":
                stack.push(stack.pop()+stack.pop());
                break;
                case "-":
                stack.push(-(stack.pop()-stack.pop()));
                break;
                case "/":
                stack.push((int) ((float)1/stack.pop()*stack.pop()));
                break;
                case "*":
                stack.push(stack.pop() * stack.pop());
                break;
                default:
                stack.push(Integer.parseInt(c));
            }
    }

    return stack.pop();
}
}

File: Data Structures & Algorithms/daily-temperatures/submission-3.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        const stack = [];
        const ans = new Array(temperatures.length).fill(0);

        for(let i = 0;i<temperatures.length;i++){
            const t = temperatures[i];
            while(stack.length && stack.at(-1)[0]<t){
                const temp = stack.pop();
                ans[temp[1]] = i - temp[1];
            }
            stack.push([t,i]);
        }

        return ans;
    }
}
